### PR TITLE
fix: Add check for parent's existence when checking delegate call

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:

Ensure that the parent frame exists when checking if the call to the parent was made as a delegate call.

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/test-clients/src/main/resource/contract/contracts/IHederaTokenService/IHederaTokenService.sol [hunk 1]
```diff
@@ -0,0 +1,766 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.4.9 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+interface IHederaTokenService {
```
Comment (IvanKavaldzhiev) [inline hedera-node/test-clients/src/main/resource/contract/contracts/IHederaTokenService/IHederaTokenService.sol:5]: We already have these resources in `resource/contract/solidity/hip-206`

**Review by iwsimon — COMMENTED** [review]
LGTM. Thanks @lukelee-sl

**Review by iwsimon — APPROVED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/HtsCallAddressChecks.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public HtsCallAddressChecks() {
      * @return true if the frame's parent is a delegate call
      */
     public boolean hasParentDelegateCall(@NonNull final MessageFrame frame) {
-        return isDelegateCall(parentOf(frame));
+        return frame.getMessageFrameStack().size() > 1 && isDelegateCall(parentOf(frame));
```
Comment (david-bakin-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/HtsCallAddressChecks.java:40]: This is fine.  Just so you know, I might have put the test over in `parentOf`, having that return an `Optional<MessageFrame`.  Or maybe have a `hasParent()` predicate.

Just a difference in coding style ....

**Review by david-bakin-sl — APPROVED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyEthereumTestsSuite.java [hunk 3]
```diff
@@ -166,6 +188,64 @@ HapiSpec legacyEtxAfterEIP155() {
...
+        final var function = getABIFor(FUNCTION, "transferToken", "IHederaTokenService");
+        final var HTS_SYSTEM_CONTRACT = "hts";
+        final var HTS_SYSTEM_CONTRACT_ADDRESS = "0000000000000000000000000000000000000167";
+
+        return propertyPreservingHapiSpec("callHtsSystemContractTest")
+                .preserving(EVM_VERSION_PROPERTY, DYNAMIC_EVM_PROPERTY)
+                .given(
+                        overriding(DYNAMIC_EVM_PROPERTY, "true"),
+                        overriding(EVM_VERSION_PROPERTY, EVM_VERSION_050),
+                        overriding(ALLOW_CALLS_TO_NON_CONTRACT_ACCOUNTS, "true"),
```
[the remark this answers is not in the document]
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyEthereumTestsSuite.java:203]: apparently not necessary.  I just added that to ensure that the test would pass.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyEthereumTestsSuite.java [hunk 4]
```diff
@@ -166,6 +188,64 @@ HapiSpec legacyEtxAfterEIP155() {
                         resetToDefault(CHAIN_ID_PROP));
     }
 
+    @HapiTest
+    final HapiSpec callHtsSystemContractTest() {
```
[the remark this answers is not in the document]
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyEthereumTestsSuite.java:192]: Well this test would not work with the change above.

**Pull request merged**